File: src/main/java/net/hectus/neobb/turn/default_game/flower/FlowerTurn.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.turn.default_game.attributes.clazz.NatureClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import org.bukkit.block.Block;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public abstract class FlowerTurn extends BlockTurn implements BuffFunction, NatureClazz {
    public FlowerTurn(NeoPlayer player) { super(player); }
    public FlowerTurn(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public boolean goodChoice(@NotNull NeoPlayer player) {
        if (player.game.history().isEmpty()) return false;
        Turn<?> last = player.game.history().getLast();
        return super.goodChoice(player) && (last instanceof TDirt || last instanceof TFlowerPot);
    }

    @Override
    public List<Buff> buffs() {
        return List.of();
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/flower/FlowerTurn.kt
package net.hectus.neobb.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.NatureClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.turn.default_game.block.BlockTurn
import org.bukkit.block.Block

abstract class FlowerTurn(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, NatureClazz {
    override fun goodChoice(player: NeoPlayer): Boolean {
        if (player.game.history.isEmpty()) return false
        val last = player.game.history.last()
        return super.goodChoice(player) && (last is TDirt || last is TFlowerPot)
    }

    override fun buffs(): List<Buff<*>> = emptyList()
}
